ngx_http_vod_module.c

```diff
@@ -3753,12 +3765,24 @@ ngx_http_vod_map_run_step(ngx_http_vod_ctx_t *ctx)
 static ngx_int_t
 ngx_http_vod_map_source_clip_done(ngx_http_vod_ctx_t *ctx)
 {
-	// initialize for reading files
-	ctx->reader = &reader_file;
-	ctx->read = (ngx_http_vod_async_read_func_t)ngx_async_file_read;
-	ctx->alloc_params_index = READER_FILE;
-	ctx->alignment = ctx->alloc_params[READER_FILE].alignment;
-	ctx->perf_counter_async_read = PC_ASYNC_READ_FILE;
+	ngx_http_vod_loc_conf_t* conf;
+	conf = ctx->submodule_context.conf;
+
+	if (conf->remote_upstream_location.len == 0)
```

Comment:
thinking about this now, if we'll use this feature in our env, we will most likely want to have it come from the json, so that we can use the same locations for both local files and remote files, but you can leave it as is for now